python/ray/dag/compiled_dag_node.py

```diff
@@ -52,22 +52,30 @@ def do_exec_compiled_task(
     self._dag_cancelled = False
 
     try:
-        self._input_channels = [i for i in inputs if isinstance(i, Channel)]
+        self._input_channels = list(
+            set(filter(lambda x: isinstance(x, Channel), inputs[0]))
+        )
+        assert (
+            len(self._input_channels) <= 1
+        ), f"Got {len(self._input_channels)} input channels, expecting no more than 1"
```

Comment:
Hmm why this limitation? We want to be able to support DAGs like:

x = f.remote()
y = f.remote()
z = g.remote(x, y)

(if this isn't already tested we should add it)

Reply:
I think this is only for the entry point of a dag, given we are packing everything into a single channel we should expect a single channel here. I updated the tests a bit to cover the case mentioned here, multi internal operands should not reach this legalization

Reply:
I see, that was a bit confusing to me. Can you:
- add a unit test for a case like the one mentioned? I don't see one in the current test suite
- update the docstring for the `inputs` arg?

Reply:
I'm trying to use [this](https://github.com/ray-project/ray/pull/42881/files#diff-7d31d4a5b6c4ac715f4d0fb1d96410a11a825cbec92f98a712835e67dc2ec4bcR57) to imitate the scenario above, glad to further update if I can have some suggestions on what is missing. docstring for inputs updated.

Reply:
Looks like this case is not working as expected. Pushed a unit test that shows the behavior.